kpi/views/v2/collection_permission_assignment.py

```diff
@@ -184,7 +183,7 @@ def clone(self, request, *args, **kwargs):
         source_collection = get_object_or_404(Collection, uid=source_collection_uid)
         user = request.user
 
-        if user.has_perm(PERM_SHARE_COLLECTION, self.collection) and \
+        if ObjectPermissionHelper.user_can_share(self.collection, user) and \
```

Comment:
I would have left a comment about `Black` syntax, but those lines are going to disappear soon ;-)